src/legacy/store/GraphQLStoreQueryResolver.js

```diff
@@ -246,8 +243,8 @@ class GraphQLStoreSingleQueryResolver {
       if (subscribedIDs) {
         // always subscribe to the root ID
         subscribedIDs[nextID] = true;
-        this._subscription = GraphQLStoreChangeEmitter.addListenerForIDs(
-          Object.keys(subscribedIDs),
+        this._subscription = this._storeData.getChangeEmitter().addListenerForIDs(
+          Object.keys(subscribedIDs).map(id => this._getCanonicalID(id)),
```

Comment:
Completely agreed, but let's change one thing at a time. First global -> contextual and make sure that's all good, then we can clean up the API.